File: python/03-binarySearch/binarysearch.py
def binary_search(arr, target):
    start = 0
    end = len(arr) -1
    while start <= end:
        mid = (start + end) // 2
        if arr[mid] < target:
            start = mid + 1
        elif arr[mid] > target:
            end = mid -1
        else:
            return mid
    return -1


if __name__ == "__main__":
    arr = [12, 24, 13, 40, 35, 26]
    target = 13
    print(binary_search(arr, target))

File: javascript/03-binary-search/BinarySearch.js
function binarySearch(arr, target){
    let start = 0
    let end = arr.length -1

    while(start <= end){
        let mid = Math.floor(start + (end - start) / 2)
        if (target < arr[mid]){
            end = mid -1;
        }
        else if (target > arr[mid]){
            start = mid + 1
        }
        else{
            return mid
        }
        
    }
     
    return -1;
}

let arr = [2, 4, 6, 8, 10, 12, 14, 16, 18, 20];
let target = 12;
console.log(binarySearch(arr, target));
